### PR TITLE
Add caching to BaseTrace.get_frequencies()

### PR DESCRIPTION
**Pull request by fschlueter**
Not that this was a bottle neck but why not saving some computing time. A small test with a 2048 sample waveforms improved the speed by a factor of 10 from 1.9us to 180ns.

### CODE REVIEW

File: NuRadioReco/framework/base_trace.py [hunk 1]
```diff
@@ -361,3 +376,7 @@ def __truediv__(self, x):
             raise ValueError('Cant divide baseTrace by number because no value is set for trace.')
         else:
             raise TypeError('Division of baseTrace object with object of type {} is not defined'.format(type(x)))
+
+@functools.lru_cache(maxsize=1024)
+def get_frequencies(length, sampling_rate):
+    return np.fft.rfftfreq(length, d=1. / sampling_rate)
```
Comment (fschlueter) [inline NuRadioReco/framework/base_trace.py:382]: This.

**Comment by fschlueter** [comment]
This PR is based on #734 and #733 . Hence I am highlighted which changes are here introduced

File: NuRadioReco/framework/base_trace.py (hunk 1 above)
Comment (shallmann) [reply NuRadioReco/framework/base_trace.py:382]: That looks good to me!

**Review by shallmann — APPROVED** [review]
The caching function looks good to me

**Pull request merged**